File: wallet-domain/src/main/java/com/example/wallet/domain/wallet/InsufficientBalanceException.java
package com.example.wallet.domain.wallet;

import com.example.wallet.domain.shared.Money;

public class InsufficientBalanceException extends RuntimeException {
    private final WalletId walletId;
    private final Money requested;
    private final Money available;

    public InsufficientBalanceException(WalletId walletId, Money requested, Money available) {
        super("insufficient balance: walletId=" + walletId + " requested=" + requested + " available=" + available);
        this.walletId = walletId;
        this.requested = requested;
        this.available = available;
    }

    public WalletId walletId() { return walletId; }
    public Money requested() { return requested; }
    public Money available() { return available; }
}

File: billing-domain/src/main/kotlin/com/example/billing/domain/wallet/InsufficientBalanceException.kt
package com.example.billing.domain.wallet

import com.example.billing.domain.shared.Money

/**
 * Wallet 잔액 부족 예외. 도메인이 직접 발행하므로 data class 가 아닌 일반 class.
 * `@get:JvmName` 으로 record-style accessor (`walletId()` / `requested()` / `available()`)
 * 보존 — Java 호출자 (RefundService, WalletQueryService 등) 무변경 동작.
 */
class InsufficientBalanceException(
    @get:JvmName("walletId") val walletId: WalletId,
    @get:JvmName("requested") val requested: Money,
    @get:JvmName("available") val available: Money,
) : RuntimeException(
    "insufficient balance: walletId=$walletId requested=$requested available=$available",
)
